controller/config_test.go

```diff
@@ -121,6 +121,27 @@ var validateTests = []struct {
 		controller.CACertKey:         testing.CACert,
 	},
 	expectError: `invalid identity public key: wrong length for base64 key, got 3 want 32`,
+}, {
+	about: "invalid management space name",
+	config: controller.Config{
+		controller.CACertKey:           testing.CACert,
+		controller.JujuManagementSpace: ` `,
+	},
+	expectError: `invalid juju mgmt space name`,
+}, {
+	about: "invalid management space name",
+	config: controller.Config{
+		controller.CACertKey:           testing.CACert,
+		controller.JujuManagementSpace: "Space",
+	},
+	expectError: `invalid juju mgmt space name`,
+}, {
+	about: "invalid management space name",
+	config: controller.Config{
+		controller.CACertKey:           testing.CACert,
+		controller.JujuManagementSpace: "\n",
```

Comment:
You didn't add a test for values that aren't strings.
And you didn't fix the extra blank spaces in the return values.